source/sav/Sav1.cpp

```diff
@@ -326,6 +341,12 @@ namespace pksm
             }
             static_cast<PK1*>(pk1.get())->boxLevel(pk1->level());
 
+            if (slot >= boxCount(box))
+            {
+                slot = boxCount(box);
+                boxCount(box, slot + 1);
+            }
```

Comment:
I don't quite like this. Instead could you just emptyPkm everything from the original box count to slot, then set boxCount to slot+1?

Reply:
See the Discord messages for why I think that idea may not work.